generate_operations.py

```diff
@@ -0,0 +1,318 @@
+#!/bin/python3
+
+import numpy as np
+import random
+
+# Settings to generate the tests
+OPERATIONS_PER_SIZE = 1
+SIZES_TO_TEST = [2, 8, 16, 22, 32]
+
+#
+# Generates a random vector of integers.
+#
+def generate_vector_elements(size):
+    return np.random.randint(0, 9, size)
+
+#
+# Returns the string that represents that vector in form of cpp array 
+# creation: '{e1, e2, e3, ...}'
+#
+def vector_string_from_array(vector):
+    elements_string = np.array2string(vector, separator=',', threshold=1000000)
+    elements_string = elements_string.replace('[', '{')
+    elements_string = elements_string.replace(']', '}')
+
+    return elements_string
+
+#
+# Creates a definition string of cpp given a name and a vector.
+#
+def vector_definition_from_array(name, vector):
+    vector_string = "vecf<{}> {}= ".format(len(vector), name)
+    elems = vector_string_from_array(vector)
+    vector_string = vector_string + elems + ';'
+    return vector_string
+
+def matrix_definition_from_array(name, nrow, ncol, vector):
+    matrix_string = "matf<{}, {}> {} = ".format(ncol, nrow, name)
+    elems = vector_string_from_array(vector)
+    matrix_string = matrix_string + elems + ';'
+    return matrix_string
+
+def print_vector_comparison_tests():
+    print('void test_vector_comparisons() {')
+    var_num = 0
+    for i in SIZES_TO_TEST:
+        for j in range(OPERATIONS_PER_SIZE):
+            a1_name = "a{}".format(var_num)
+            b1_name = "b{}".format(var_num)
```

Comment:
The variable name  `a1_name` and `b1_name` should just be `a_name` and `b_name`. Not only here, but in all the tests.

Reply:
Fixed on latest commit.